test_flush_connection_uds randomly fails with BrokenPipeError

```
__________________________ test_flush_connection_uds ___________________________

endpoint_uds_server = <tests.test_api.UDSHTTPServer object at 0x7fb499ad54a8>

    def test_flush_connection_uds(endpoint_uds_server):
        payload = mock.Mock()
        payload.get_payload.return_value = 'foobar'
        payload.length = 12
        api = API(_HOST, 2019, uds_path=endpoint_uds_server.server_address)
        response = api._flush(payload)
>       assert response.status == 200
E       AttributeError: 'BrokenPipeError' object has no attribute 'status'

tests/test_api.py:267: AttributeError
=============================== warnings summary ===============================
```

https://app.circleci.com/jobs/github/DataDog/dd-trace-py/193358/parallel-runs/0/steps/0-104

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -10,8 +10,8 @@ from unittest import TestCase
 
 import pytest
 
-from ddtrace.api import API, Response
-from ddtrace.compat import iteritems, httplib, PY3
+from ddtrace.api import API, Response, UDSHTTPConnection
+from ddtrace.compat import iteritems, httplib, PY3, get_connection_response
 from ddtrace.internal.runtime.container import CGroupInfo
 from ddtrace.vendor.six.moves import BaseHTTPServer, socketserver
 
@@ -61,7 +61,16 @@ def _make_uds_server(path, request_handler):
     t.start()
 
     # Wait for the server to start
-    time.sleep(0.2)
+    resp = None
+    while resp != 200:
+        conn = UDSHTTPConnection(server.server_address, False, _HOST, 2019)
+        try:
+            conn.request("PUT", path)
+            resp = get_connection_response(conn).status
+        finally:
+            conn.close()
+        time.sleep(0.01)
+
     return server, t
 
 

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -59,6 +59,9 @@ def _make_uds_server(path, request_handler):
     # Set daemon just in case something fails
     t.daemon = True
     t.start()
+
+    # Wait for the server to start
+    time.sleep(0.2)
     return server, t
 
 

